lib/view/solve_crossword.dart

```diff
@@ -221,9 +271,100 @@ class TakePictureScreenState extends State<TakePictureScreen> {
     crosswordInfoRepository.insertCrosswordInfo(crosswordInfo);
   }
 
-  void navigateToApp(context) {
-    Navigator.of(context).push(MaterialPageRoute(
-      builder: (context) => const App(),
-    ));
+  SnackBar serverErrorSnackBar(String statusCode) {
+    return SnackBar(
+        behavior: SnackBarBehavior.floating,
+        backgroundColor: Colors.red,
+        content: Text("Error! Status code: $statusCode"));
+  }
+
+  Future<dynamic> checkCrosswordStatus(
+      String userId, String crosswordId, BuildContext context) async {
+    print("checkCrosswordStatus");
+
+    var resposne =
+        await HttpUtil.getCrosswordStatus(userId, crosswordId.toString());
+
+    var decodedResponse = jsonDecode(resposne.body);
+    print(decodedResponse);
+
+    var status = decodedResponse['status'];
+    print(status);
+
+    while (status != "solved_waiting" && status != "cannot_solve") {
+      await Future.delayed(const Duration(seconds: 1));
+
+      var resposne =
+          await HttpUtil.getCrosswordStatus(userId, crosswordId.toString());
+
+      decodedResponse = jsonDecode(resposne.body);
+      print(decodedResponse);
+
+      status = decodedResponse['status'];
+      print(status);
+    }
+
+    if (status == "cannot_solve") {
+      print("cant solve crossword");
+    } else {
+      print("crossword solved - waiting");
+    }
+
+    return decodedResponse;
+  }
+
+  SnackBar serverSolvedMessageSnackbar(
+      context, dynamic decodedResponse, String path, String id, String name) {
+    var status = decodedResponse['status'];
+    var message = decodedResponse['message'];
+
+    return SnackBar(
+      behavior: SnackBarBehavior.floating,
+      content: Text("Status: $status, message: $message"),
+      backgroundColor: (Colors.green),
+      action: SnackBarAction(
+        label: 'Go to crossword',
```

Comment:
```suggestion
      content: Text("Pomyślnie rozwiązano krzyżówkę"),
      backgroundColor: (Colors.green),
      action: SnackBarAction(
        label: 'Pokaż',
```
And change something with this blue text. Maybe the good idea is to set bold on this text?